### PR TITLE
Rewrite internals with ES6

### PR DESCRIPTION
**Pull request by tausch86**
@zenoamaro, Thanks for this quill wrapper! I saw that the project is kind of stalled out right now and took it upon myself to refactor the wrapper to be ES6 class based. Not expecting this to be merged, more interested in your feedback and if I should maintain this fork separately as I'm sure there's a lot of work to do in here. 

Also, note that the demo is functioning

**Changes:**
* Refactored `react.createClass()` to `class ... extends Component` and various other syntactic sugars 
* Put `mixin` into `Quill Component`
* Updated demo example to also follow es6
* Created Demo webpack config
* Added babel and some presets 
* Added Utils file for misc constants and helpers
* Also changed tabs to spaces
* removed `quill` folder

### CODE REVIEW

File: src/toolbar.js [hunk 1]
```diff
@@ -1,179 +1,131 @@
...
-			dangerouslySetInnerHTML: { __html:html }
-		});
-	}
-
-});
-
-module.exports = QuillToolbar;
-QuillToolbar.defaultItems = defaultItems;
-QuillToolbar.defaultColors = defaultColors;
+import React, { Component, PropTypes } from 'react'
```
Comment (vladbytex) [inline src/toolbar.js]: Please consider using separate [prop-types](https://github.com/facebook/prop-types) package instead using PropTypes from React. PropTypes from React will be deprecated in react@16.0.

Comment (alexkrolick) [reply src/toolbar.js]: This PR is substantially out of date, but we are keeping it open as a reference for future  work. We have already moved to React 16-compatible utilities in the main repo.

**Comment by jonathanlaf** [comment]
It this pull compatible with React 16 ?

**Comment by alexkrolick** [comment]
@jonathanlaf see https://github.com/zenoamaro/react-quill/pull/129#discussion_r141374517

**Comment by alexkrolick** [comment]
@jonathanlaf  Please open an issue outside this PR. This code is a proposed rewrite of the module _internals_ and doesn't affect the API - the `master` branch/NPM version should work just fine with React 16.

**Comment by zenoamaro** [comment]
This has been fixed with the recent beta release! 

Please make sure that it fits your needs, or open an issue.

Thanks for the wait!

**Pull request closed without merging**